networkmanager/networkmanager_test.go

```diff
@@ -0,0 +1,334 @@
+				Instance:  2,
+			},
+		},
+	}
+
+	for _, data := range testData {
+		if data.removeConfig {
+			manager.RemoveInstanceNetworkParameters(data.instance, "network1")
+
+			continue
+		}
+
+		networkParameters, err := manager.PrepareInstanceNetworkParameters(data.instance, "network1")
+		if err != nil {
+			t.Fatalf("Can't prepare instance network configuration: %v", err)
+		}
+
+		if networkParameters != data.networkParameters {
+			t.Errorf("Wrong network parameters: %v", data.networkParameters)
+		}
+	}
+
+	expectedInstancesIdent := []aostypes.InstanceIdent{
+		{
+			ServiceID: "service1",
+			SubjectID: "subject1",
+			Instance:  1,
+		},
+		{
+			ServiceID: "service1",
+			SubjectID: "subject1",
+			Instance:  2,
+		},
+	}
+
+	instances := manager.GetInstances()
+	if !compareInstancesIdent(instances, expectedInstancesIdent) {
+		t.Error("Unexpected instances ident")
+	}
+}
+
+func TestNetworkStorage(t *testing.T) {
+	ipam := ipamTest{}
+	if err := ipam.init(); err != nil {
+		t.Fatalf("Can't init ipam management: %v", err)
+	}
+
+	networkmanager.GetIPSubnet = ipam.getIPSubnet
+
+	storage := &testStore{
+		networkInfos: make(map[aostypes.InstanceIdent]networkmanager.NetworkInfo),
+	}
+
+	manager, err := networkmanager.New(storage)
+	if err != nil {
+		t.Fatalf("Can't create network manager: %v", err)
+	}
+
+	testData := []struct {
+		networkParameters aostypes.NetworkParameters
+		instance          aostypes.InstanceIdent
+	}{
+		{
+			networkParameters: aostypes.NetworkParameters{
+				IP:     ("172.17.0.2"),
+				Subnet: ("172.17.0.0/16"),
+			},
+			instance: aostypes.InstanceIdent{
+				ServiceID: "service1",
+				SubjectID: "subject1",
+				Instance:  1,
+			},
+		},
+		{
+			networkParameters: aostypes.NetworkParameters{
+				IP:     ("172.17.0.3"),
+				Subnet: ("172.17.0.0/16"),
+			},
+			instance: aostypes.InstanceIdent{
+				ServiceID: "service1",
+				SubjectID: "subject1",
+				Instance:  2,
+			},
+		},
+	}
+
+	for _, data := range testData {
+		if _, err := manager.PrepareInstanceNetworkParameters(
+			data.instance, "network1"); err != nil {
+			t.Fatalf("Can't prepare instance network configuration: %v", err)
+		}
+	}
+
+	manager1, err := networkmanager.New(storage)
+	if err != nil {
+		t.Fatalf("Can't create network manager: %v", err)
+	}
+
+	expectedInstancesIdent := []aostypes.InstanceIdent{
+		{
+			ServiceID: "service1",
+			SubjectID: "subject1",
+			Instance:  1,
+		},
+		{
+			ServiceID: "service1",
+			SubjectID: "subject1",
+			Instance:  2,
+		},
+	}
+
+	instances := manager1.GetInstances()
+	if !compareInstancesIdent(instances, expectedInstancesIdent) {
+		t.Error("Unexpected instances ident")
+	}
+}
+
+/***********************************************************************************************************************
+ * Interfaces
+ **********************************************************************************************************************/
+
+func (ipam *ipamTest) init() error {
```

Comment:
create new function which will initialize and create ipamTest instance.

Reply:
Done